sys/boot/i386/gptboot/gptboot.c

```diff
@@ -251,7 +252,8 @@ gptinit(void)
 #ifdef LOADER_GELI_SUPPORT
 	if (geli_taste(vdev_read, &dsk, (gpttable[curent].ent_lba_end -
 	    gpttable[curent].ent_lba_start)) == 0) {
-		if (geli_passphrase(&gelipw, dsk.unit, 'p', curent + 1, &dsk) != 0) {
+		if (geli_havekey(&dsk) != 0 && geli_passphrase(&gelipw,
```

Comment:
don't prompt for a password if we already have a key.

Both of these functions call the internal geli_attach() once they have a key.